zerver/views/users.py

```diff
@@ -468,24 +468,27 @@ def regenerate_bot_api_key(
 
 
 @require_member_or_admin
-@has_request_variables
+@typed_endpoint
 def add_bot_backend(
     request: HttpRequest,
     user_profile: UserProfile,
-    full_name_raw: str = REQ("full_name"),
-    short_name_raw: str = REQ("short_name"),
-    bot_type: int = REQ(json_validator=check_int, default=UserProfile.DEFAULT_BOT),
-    payload_url: str = REQ(json_validator=check_url, default=""),
-    service_name: Optional[str] = REQ(default=None),
-    config_data: Mapping[str, str] = REQ(
-        default={}, json_validator=check_dict(value_validator=check_string)
-    ),
-    interface_type: int = REQ(json_validator=check_int, default=Service.GENERIC),
-    default_sending_stream_name: Optional[str] = REQ("default_sending_stream", default=None),
-    default_events_register_stream_name: Optional[str] = REQ(
-        "default_events_register_stream", default=None
-    ),
-    default_all_public_streams: Optional[bool] = REQ(json_validator=check_bool, default=None),
+    *,
+    full_name_raw: Annotated[str, ApiParamConfig("full_name")],
+    short_name_raw: Annotated[str, ApiParamConfig("short_name")],
+    bot_type: Json[int] = UserProfile.DEFAULT_BOT,
+    payload_url: Json[
+        Annotated[str, AfterValidator(lambda x: check_param_url("payload_url", x))]
```

Comment:
Is there a better way to do this `AfterValidator` pattern that woudln't require the `lambda x` wrapper?

Reply:
For validation functions that accept only 1 positional argument we could pass the function directly to `AfterValidator`. But for functions like `check_int_in` which take a second parameter we would need tot use `lambda` or a similar wrapper function. Both `lambda` and wrapper functions have been used in the docs, the do the same thing esssentially.